interface/VideoCommonDefs.h

```diff
@@ -21,30 +21,31 @@
 #include <stdint.h>
 
 #ifdef __cplusplus
-#ifdef ANDROID
-#include <memory>
-#else
-#include <tr1/memory>
-#endif
-namespace YamiMediaCodec{
-#ifdef ANDROID
-    #define SharedPtr std::shared_ptr
-    #define WeakPtr std::weak_ptr
-    #define DynamicPointerCast std::dynamic_pointer_cast
-    #define StaticPointerCast std::static_pointer_cast
-    #define EnableSharedFromThis std::enable_shared_from_this
-#else
-    #define SharedPtr std::tr1::shared_ptr
-    #define WeakPtr std::tr1::weak_ptr
-    #define DynamicPointerCast std::tr1::dynamic_pointer_cast
-    #define StaticPointerCast std::tr1::static_pointer_cast
-    #define EnableSharedFromThis std::tr1::enable_shared_from_this
-#endif
 
-#define SharedFromThis shared_from_this
-
-}
-#endif
+#if __cplusplus > 199711L
+    #include <memory>
+#else
+    #include <tr1/memory>
+    namespace std {
+
+        #if __cplusplus < 201103L
```

Comment:
Hmmm... odd.  I removed them and now it compiles fine on 6.x.  I must have been in an intermediate change state when I encountered problems.  I will remove this check in an update.